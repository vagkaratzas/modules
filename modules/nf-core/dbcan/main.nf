process DBCAN {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/dbcan:4.1.4--pyhdfd78af_1':
        'biocontainers/dbcan:4.1.4--pyhdfd78af_1' }"

    input:
    tuple val(meta), path(faa)

    output:
    tuple val(meta), path("dbsub.out"), emit: dbsub
    path "versions.yml"               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    run_dbcan \\
        $args \\
        ${faa} \\
        protein \\
        -out_dir ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: \$(dbcan --help)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}
