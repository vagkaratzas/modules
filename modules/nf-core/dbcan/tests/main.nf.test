nextflow_process {

    name "Test Process DBCAN"
    script "../main.nf"
    process "DBCAN"

    tag "modules"
    tag "modules_nfcore"
    tag "dbcan"

    test("ecoli - fna") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/proteome.fasta', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
