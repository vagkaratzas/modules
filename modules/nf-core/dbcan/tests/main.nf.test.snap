{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "dbsub": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.5"
        },
        "timestamp": "2024-10-16T14:05:05.507427642"
    },
    "ecoli - fna": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "dbsub": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.5"
        },
        "timestamp": "2024-10-16T14:04:58.313910313"
    }
}